cluster/aws/util.sh

```diff
@@ -1255,57 +1357,77 @@ function start-minions() {
     echo "API_SERVERS: $(yaml-quote ${MASTER_INTERNAL_IP:-})"
     echo "__EOF_KUBE_ENV_YAML"
     echo ""
-    echo "wget -O bootstrap ${BOOTSTRAP_SCRIPT_URL}"
+    if [[ $PRESET_INSTANCES_ENABLED != $TRUE ]]; then
+      echo "wget -O bootstrap ${BOOTSTRAP_SCRIPT_URL}"
+    else
+      echo "cp ${BOOTSTRAP_SCRIPT_URL} bootstrap"
+    fi
     echo "chmod +x bootstrap"
     echo "mkdir -p /etc/kubernetes"
     echo "mv kube_env.yaml /etc/kubernetes"
     echo "mv bootstrap /etc/kubernetes/"
-    echo "cat > /etc/rc.local << EOF_RC_LOCAL"
-    echo "#!/bin/sh -e"
-    # We want to be sure that we don't pass an argument to bootstrap
-    echo "/etc/kubernetes/bootstrap"
-    echo "exit 0"
-    echo "EOF_RC_LOCAL"
-    echo "/etc/kubernetes/bootstrap"
+    if [[ $PRESET_INSTANCES_ENABLED != $TRUE ]]; then
+      echo "cat > /etc/rc.local << EOF_RC_LOCAL"
+      echo "#!/bin/sh -e"
+      # We want to be sure that we don't pass an argument to bootstrap
+      echo "/etc/kubernetes/bootstrap"
+      echo "exit 0"
+      echo "EOF_RC_LOCAL"
+      echo "/etc/kubernetes/bootstrap > /tmp/bootstrap.log"
+    fi
   ) > "${KUBE_TEMP}/node-user-data"
 
-  # Compress the data to fit under the 16KB limit (cloud-init accepts compressed data)
-  gzip "${KUBE_TEMP}/node-user-data"
+  if [[ $PRESET_INSTANCES_ENABLED != $TRUE ]]; then
+    # Compress the data to fit under the 16KB limit (cloud-init accepts compressed data)
+    gzip "${KUBE_TEMP}/node-user-data"
 
-  local public_ip_option
-  if [[ "${ENABLE_NODE_PUBLIC_IP}" == "true" ]]; then
-    public_ip_option="--associate-public-ip-address"
-  else
-    public_ip_option="--no-associate-public-ip-address"
-  fi
-  local spot_price_option
-  if [[ -n "${NODE_SPOT_PRICE:-}" ]]; then
-    spot_price_option="--spot-price ${NODE_SPOT_PRICE}"
+    local public_ip_option
+    if [[ "${ENABLE_NODE_PUBLIC_IP}" == "true" ]]; then
+      public_ip_option="--associate-public-ip-address"
+    else
+      public_ip_option="--no-associate-public-ip-address"
+    fi
+    local spot_price_option
+    if [[ -n "${NODE_SPOT_PRICE:-}" ]]; then
+      spot_price_option="--spot-price ${NODE_SPOT_PRICE}"
+    else
+      spot_price_option=""
+    fi
+    ${AWS_ASG_CMD} create-launch-configuration \
```

Comment:
This is not new code but git thought it was all new. It usually does not do it.

Reply:
It's github diff issue. There is a setting to avoid such issue. In this page, there is a setting button, check "hide whitespace changes", click "Apply and reload" button. Then the issue will gone.

Reply:
It's github diff issue. There is a setting to avoid such issue. In this page, there is a setting button, check "hide whitespace changes", click "Apply and reload" button. Then the issue will gone.